healthCheck data race

```
t=2019-03-11T14:13:46+0000 lvl=info msg="healthCheck start " addr=localhost:8805 inter=1 times=2
t=2019-03-11T14:13:47+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=false
t=2019-03-11T14:13:47+0000 lvl=info msg="healthCheck listen open"
t=2019-03-11T14:13:48+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=true
t=2019-03-11T14:13:49+0000 lvl=info msg="healthCheck start " addr=localhost:8805 inter=1 times=2
t=2019-03-11T14:13:49+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=true
t=2019-03-11T14:13:50+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=true
t=2019-03-11T14:13:50+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=false
t=2019-03-11T14:13:50+0000 lvl=eror msg="healthCheck " listen open err="listen tcp 127.0.0.1:8805: bind: address already in use"
t=2019-03-11T14:13:51+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=true
t=2019-03-11T14:13:51+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=false
t=2019-03-11T14:13:51+0000 lvl=eror msg="healthCheck " listen open err="listen tcp 127.0.0.1:8805: bind: address already in use"
t=2019-03-11T14:13:52+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=true
t=2019-03-11T14:13:52+0000 lvl=info msg="healthCheck tick" peers=2 isCaughtUp=true health=true listen=false
==================
WARNING: DATA RACE
Write at 0x00c4200686d0 by goroutine 18:
  github.com/33cn/chain33/util.(*HealthCheckServer).listen()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck.go:76 +0xcc
  github.com/33cn/chain33/util.(*HealthCheckServer).healthCheck()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck.go:136 +0x247
Previous read at 0x00c4200686d0 by goroutine 16:
  github.com/33cn/chain33/util.(*HealthCheckServer).healthCheck()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck.go:121 +0x50b
Goroutine 18 (running) created at:
  github.com/33cn/chain33/util.(*HealthCheckServer).Start()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck.go:66 +0x25c
  github.com/33cn/chain33/util.TestStart()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck_test.go:36 +0x84b
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.7.linux.amd64/src/testing/testing.go:746 +0x16c
Goroutine 16 (finished) created at:
  github.com/33cn/chain33/util.(*HealthCheckServer).Start()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck.go:66 +0x25c
  github.com/33cn/chain33/util.TestStart()
      /home/travis/gopath/src/github.com/33cn/chain33/util/healthcheck_test.go:33 +0x685
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.7.linux.amd64/src/testing/testing.go:746 +0x16c
==================
t=2019-03-11T14:13:52+0000 lvl=info msg="healthCheck listen open"
t=2019-03-11T14:13:52+0000 lvl=info msg="healthCheck quit"
--- FAIL: TestStart (7.04s)
	testing.go:699: race detected during execution of test
t=2019-03-11T14:13:53+0000 lvl=info msg="healthCheck tick" peers=1 isCaughtUp=true health=false listen=true
t=2019-03-11T14:13:53+0000 lvl=info msg="current user data dir is " module=util dir=/tmp/chain33datadir-985992544/hello
t=2019-03-11T14:13:53+0000 lvl=info msg="current user data dir is " module=util dir=/TEMP/hello
t=2019-03-11T14:13:54+0000 lvl=info msg="current user data dir is " module=util dir=/home/travis/hello
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkBlockHash
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkBlockHash
t=2019-03-11T14:13:54+0000 lvl=dbug msg="reply msg ok" module=queue msg="{topic:mempool, Ty:EventTx, Id:1, Err:<nil>, Ch:true}"
t=2019-03-11T14:13:54+0000 lvl=dbug msg=ExecBlock module=util height------->=0 ntx=1
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkCheckTxDup
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CheckTxDup=517.081µs
t=2019-03-11T14:13:54+0000 lvl=dbug msg=ExecBlock module=util prevtx=1 newtx=1
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util ExecTx=327.262µs
t=2019-03-11T14:13:54+0000 lvl=eror msg="exec tx err" module=util err=
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CalcMerkleRootCache=322.871µs
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CheckBlock=380.951µs
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util height=0 ntx=1 writebatchsync=true cost=249.034µs
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkBlockHash
t=2019-03-11T14:13:54+0000 lvl=dbug msg=ExecBlock module=util height------->=1 ntx=0
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkCheckTxDup
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CheckTxDup=919.004µs
t=2019-03-11T14:13:54+0000 lvl=dbug msg=ExecBlock module=util prevtx=0 newtx=0
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util ExecTx=473.522µs
t=2019-03-11T14:13:54+0000 lvl=eror msg="exec tx err" module=util err=
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CalcMerkleRootCache=104.392µs
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CheckBlock=350.671µs
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util height=1 ntx=0 writebatchsync=true cost=424.563µs
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkBlockHash
t=2019-03-11T14:13:54+0000 lvl=dbug msg=ExecBlock module=util height------->=1 ntx=0
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkCheckTxDup
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CheckTxDup=347.312µs
t=2019-03-11T14:13:54+0000 lvl=dbug msg=ExecBlock module=util prevtx=0 newtx=0
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util ExecTx=13.711577ms
t=2019-03-11T14:13:54+0000 lvl=eror msg="exec tx err" module=util err=
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CalcMerkleRootCache=92.265µs
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util CheckBlock=284.884µs
t=2019-03-11T14:13:54+0000 lvl=info msg=ExecBlock module=util height=1 ntx=0 writebatchsync=true cost=250.511µs
t=2019-03-11T14:13:54+0000 lvl=eror msg="getfork title not exisit -> " module=types title= key=ForkCheckTxDup
t=2019-03-11T14:13:54+0000 lvl=dbug msg=SendErrEventToFront test=nil LOG15_ERROR="Normalized odd number of arguments by adding nil" frommodule=from tomodule=to err=test
FAIL
```

--- a/build/docker-compose.sh
+++ b/build/docker-compose.sh
@@ -300,11 +300,23 @@ function transfer() {
     done
 
     ${CLI} account balance -a 12qyocayNF7Lv6C9qW4avxs2E7U41fKSfv -e coins
-    balance=$(${CLI} account balance -a 12qyocayNF7Lv6C9qW4avxs2E7U41fKSfv -e coins | jq -r ".balance")
-    if [ "${balance}" != "10.0000" ]; then
-        echo "wrong balance=$balance, should not be 10.0000"
-        exit 1
-    fi
+
+    local times=100
+    while true; do
+        balance=$(${CLI} account balance -a 12qyocayNF7Lv6C9qW4avxs2E7U41fKSfv -e coins | jq -r ".balance")
+        echo "account balance is ${balance}, expect 10.0000 "
+        if [ "${balance}" != "10.0000" ]; then
+            block_wait 2
+            times=$((times - 1))
+            if [ $times -le 0 ]; then
+                echo "account balance transfer failed"
+                exit 1
+            fi
+        else
+            echo "account balance transfer success"
+            break
+        fi
+    done
 
 }
 

--- a/util/healthcheck.go
+++ b/util/healthcheck.go
@@ -105,7 +105,7 @@ func (s *HealthCheckServer) getHealth(sync bool) (bool, error) {
 		return false, err
 	}
 
-	log.Info("healthCheck tick", "peers", len(peerList.Peers), "isCaughtUp", reply.IsOk,
+	log.Debug("healthCheck tick", "peers", len(peerList.Peers), "isCaughtUp", reply.IsOk,
 		"health", len(peerList.Peers) > 1 && reply.IsOk, "listen", sync)
 
 	return len(peerList.Peers) > 1 && reply.IsOk, nil

--- a/util/healthcheck_test.go
+++ b/util/healthcheck_test.go
@@ -20,8 +20,8 @@ func TestStart(t *testing.T) {
 	health := NewHealthCheckServer(q.Client())
 
 	api := new(mocks.QueueProtocolAPI)
-	reply := &types.Reply{IsOk: true}
-	api.On("IsSync").Return(reply, nil)
+	api.On("IsSync").Return(&types.Reply{IsOk: true}, nil).Times(2)
+	api.On("IsSync").Return(&types.Reply{IsOk: false}, nil)
 	peer1 := &types.Peer{Addr: "addr1"}
 	peer2 := &types.Peer{Addr: "addr2"}
 	peers := &types.PeerList{Peers: []*types.Peer{peer1, peer2}}
@@ -31,12 +31,11 @@ func TestStart(t *testing.T) {
 
 	cfg, _ := types.InitCfg("../cmd/chain33/chain33.test.toml")
 	health.Start(cfg.Health)
-	time.Sleep(time.Second * 3)
-	api.On("IsSync").Return(&types.Reply{IsOk: false}, nil)
-	health.Start(cfg.Health)
-	time.Sleep(time.Second * 3)
+	time.Sleep(time.Second * 6)
+
 	health.Close()
 	time.Sleep(time.Second * 1)
+
 }
 
 func TestGetHealth(t *testing.T) {

